### PR TITLE
Smart metering

### PR DESCRIPTION
**Pull request by rlemmers**

### CODE REVIEW

**Comment by CLAAdmin** [comment]
[CLA assistant check](http://52.16.237.165:80/OSGP/Platform?pullRequest=107) <br/>All committers of the pull request should sign our Contributor License Agreement in order to get your pull request merged.<br/>**0** out of **3** committers have signed the CLA.<br/><br/>:x: kevinsmeets<br/>:x: Sander3003<br/>:x: rlemmers

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/OsgpCoreResponseMessageProcessor.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import com.alliander.osgp.shared.infra.jms.ResponseMessageResultType;
+
+/**
+ * Base class for MessageProcessor implementations. Each MessageProcessor
+ * implementation should be annotated with @Component. Further the MessageType
+ * the MessageProcessor implementation can process should be passed in at
+ * construction. The Singleton instance is added to the HashMap of
+ * MessageProcessors after dependency injection has completed.
+ *
+ * @author CGI
```
Comment (kevinsmeets) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/OsgpCoreResponseMessageProcessor.java]: Please remove the author CGI from this comment.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/ws/WebServiceRequestMessageProcessor.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import com.alliander.osgp.shared.infra.jms.ResponseMessageResultType;
+
+/**
+ * Base class for MessageProcessor implementations. Each MessageProcessor
+ * implementation should be annotated with @Component. Further the MessageType
+ * the MessageProcessor implementation can process should be passed in at
+ * construction. The Singleton instance is added to the HashMap of
+ * MessageProcessors after dependency injection has completed.
+ *
+ * @author CGI
```
Comment (kevinsmeets) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/ws/WebServiceRequestMessageProcessor.java]: Please remove the author CGI from this comment.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/infra/jms/SmartMeteringRequestMessageSender.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jms.core.JmsTemplate;
+import org.springframework.jms.core.MessageCreator;
+
+import com.alliander.osgp.shared.infra.jms.Constants;
+
+/**
+ * Class for sending smart metering request messages to a queue
+ *
+ * @author CGI
```
Comment (kevinsmeets) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/infra/jms/SmartMeteringRequestMessageSender.java]: Please remove the author CGI from this comment.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/infra/jms/SmartMeteringRequestMessageType.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.alliander.osgp.adapter.ws.smartmetering.infra.jms;
+
+/**
+ * Enumeration of smart metering request message types
+ *
+ * @author CGI
```
Comment (kevinsmeets) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/infra/jms/SmartMeteringRequestMessageType.java]: Please remove the author CGI from this comment.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/infra/jms/SmartMeteringResponseMessageFinder.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jms.core.JmsTemplate;
+
+import com.alliander.osgp.shared.infra.jms.BaseResponseMessageFinder;
+
+/**
+ * Class for retrieving response messages from the smart metering responses
+ * queue by correlation UID.
+ *
+ * @author CGI
```
Comment (kevinsmeets) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/infra/jms/SmartMeteringResponseMessageFinder.java]: Please remove the author CGI from this comment.

**Pull request merged**